lib/elixir/lib/module.ex

```diff
@@ -695,6 +695,21 @@ defmodule Module do
   defp simplify_signature(other, acc) when is_binary(other),  do: autogenerated(acc, :binary)
   defp simplify_signature(_, acc),                            do: autogenerated(acc, :arg)
 
+  defp simplify_struct(module) when is_atom(module) do
```

Comment:
I think this can be written like this:

```elixir
try do
  split(module)
rescue
  ArgumentError -> module
else
  struct_name -> String.to_atom(camelcase_to_underscore(List.last(struct_name)))
end
```